### PR TITLE
doc: note deprecation of Cache Tiering in Reef

### PR DESCRIPTION
**Pull request by zdover23**
Add a note to the documentation that cache tiering is deprecated in Reef.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -1094,6 +1094,8 @@ See `Erasure Code Notes`_ for additional details.
 Cache Tiering
 -------------
 
+.. note:: Cache tiering is deprecated in Reef.
```
Comment (anthonyeleven) [inline doc/architecture.rst:1097]: Almost wonder if s/is/was/ for futureproofing.

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:1097]: I think that we can assume the historical present for the sake of posterity.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51678 - Reef backport

I don't _think_ it's necessary to backport this to other release branches at this time, but I am willing to hear the argument that it is.

**Comment by anthonyeleven** [comment]
Arguably if someone doesn't already know they haven't been paying attention.

**Pull request merged**